Etities/Refrigerator.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RefrigeratorExercise.Etities
+{
+    public class Refrigerator
+    {
+        private static int nextRefrigeratorId = 0;
+        private List<Refrigerator> refrigerators;
+
+        public int RefrigeratorId { get; }
+        public string RefrigeratorModel { get; set; }
+        public string RefrigeratorColor { get; set; }
+        public int NumOfShelves { get; set; }
+        public List<Shelf> Shelves { get; set; }
+
+        public Refrigerator()
+        {
+            RefrigeratorId = ++nextRefrigeratorId;
+            refrigerators = new List<Refrigerator>();
+        }
+        public override string ToString()
+        {
+            return $"Refrigerator ID: {RefrigeratorId}\nModel: {RefrigeratorModel}\nColor: {RefrigeratorColor}\nNumber of Shelves: {NumOfShelves}";
+        }
+        public double CalculateRemainingSpaceInRefrigerator()
+        {
+            double totalSpace = 0;
+            foreach (Shelf shelf in Shelves)
+            {
+                totalSpace += shelf.SpaceInTheShelf();
+            }
+            Console.WriteLine( $"total:{totalSpace}");
+            return totalSpace;
+        }
+        public void AddItemToRefrigerator(Item item)
+        {
+            foreach (Shelf shelf in Shelves)
+            {
+                if (shelf.SpaceInTheShelf() >= item.ItemSpace)
+                { 
+                    shelf.Items.Add(item);
+                    item.IdOfShelf = shelf.ShelfId;                 
+                    Console.WriteLine("item added successfuly");
+                    return;
+                }
+            }
+            Console.WriteLine("there is no space in the refrigeratot");
+        }
+        public Item RemoveItemToRefrigerator(int id)
+        {
+            foreach (Shelf shelf in Shelves)
+            {
+                foreach(Item item in shelf.Items)
+                {
```

Comment:
You are implementing logic that should be in Shelf in the Refrigerator class. this would not be possible if you had encapsulated fields properly.